odoo/http.py

```diff
@@ -1,1710 +1,1589 @@
+                        raise  # bubble up to odoo.http.Application.__call__
+                    if 'werkzeug' in config['dev_mode']:
+                        raise  # bubble up to werkzeug.debug.DebuggedApplication
+
+                    if request.type == 'json':
+                        exc.error_response = self.env['ir.http']._json_handle_error(exc)
                     else:
-                        result = ir_http._dispatch()
-                else:
-                    result = _dispatch_nodb()
-
-                response = self.get_response(httprequest, result, explicit_session)
-            return response(environ, start_response)
-
-        except werkzeug.exceptions.HTTPException as e:
-            return e(environ, start_response)
-
-    def get_profiler_context_manager(self, request):
-        """ Return a context manager that combines a profiler and ``request``. """
-        if request.session.profile_session and request.session.db:
-            if request.session.profile_expiration < str(datetime.now()):
-                # avoid having session profiling for too long if user forgets to disable profiling
-                request.session.profile_session = None
-                _logger.warning("Profiling expiration reached, disabling profiling")
-            elif 'set_profiling' in request.httprequest.path:
-                _logger.debug("Profiling disabled on set_profiling route")
-            elif request.httprequest.path.startswith('/longpolling'):
-                _logger.debug("Profiling disabled for longpolling")
-            elif odoo.evented:
-                # only longpolling should be in a evented server, but this is an additional safety
-                _logger.debug("Profiling disabled for evented server")
-            else:
-                try:
-                    prof = profiler.Profiler(
-                        db=request.session.db,
-                        description=request.httprequest.full_path,
-                        profile_session=request.session.profile_session,
-                        collectors=request.session.profile_collectors,
-                        params=request.session.profile_params,
-                    )
-                    return profiler.Nested(prof, request)
-                except Exception:
-                    _logger.exception("Failure during Profiler creation")
-                    request.session.profile_session = None
-        return request
+                        exc.error_response = self.env['ir.http']._http_handle_error(exc)
+                    raise
 
-    def get_db_router(self, db):
-        if not db:
-            return self.nodb_routing_map
-        return request.registry['ir.http'].routing_map()
+    def _serve_ir_http(self):
+        """
+        Delegate most of the processing to the ir.http model that is
+        extensible by applications.
+        """
+        ir_http = self.env['ir.http']
```

Comment:
All the invoked methods are class methods, so getting a recordset to invoke them is simply useless.
```suggestion
        ir_http = self.registry['ir.http']
```